Arguments parsing implementation

shpingalet007 opened the pull request:
## Changes
1. **Added GitHub Actions input** - `allow_arguments: Boolean`. It enables/disables arguments parsing. It is set to `false` by default.
2. **Arguments parsing via double-asterisk marker** - script looks for `**` in trigger and considers the marked part of the comment as argument. Output to the `steps.check.outputs.arguments`. It doesn't do anything if `allow_arguments: false`.
3. **Add new example** - add job that enables debugging `#deploy on ** environment` comment.

## Other changes
- Replace npm **`@zeit/ncc`** to **`@vercel/ncc`** as it was deprecated.
- Upload new action NCC build.
- Documented everything in **`Readme.md`**.

shpingalet007 committed:
Enable comment arguments flag

shpingalet007 committed:
Add arguments parsing and output them

Parsed only when allow_arguments is set

shpingalet007 committed:
New example workflow with arguments

shpingalet007 committed:
Replace @zeit/ncc to @vercel/ncc

@zeit/ncc is deprecated

shpingalet007 committed:
Build Github Action distribution

shpingalet007 committed:
Add new functionality description in readme

mattkeeler commented:
This would be a great feature to add.

eliasmelgaco commented:
hey @shpingalet007 can you publish on github marketplace as [shanegenschaw](https://github.com/shanegenschaw) did here?  https://github.com/Khan/pull-request-comment-trigger/pull/46 ?

shpingalet007 commented:
@eliasmelgaco I could do it, but wouldn't it be better if [shanegenschaw](https://github.com/shanegenschaw) merges my PR? There are already 5 modules (original + 4 forks) published on GitHub Actions Marketplace. I would recommend the community to select his repository as main fork while patches are not accepted in the original one. He already has 3 stars on his repo and looks like the most active fork. Cheers

shanegenschaw commented:
Hey, @shpingalet007 and @eliasmelgaco, I have a PR ready to pull this into my fork, please take a look/give a trial, and let me know if it looks good:

- https://github.com/shanegenschaw/pull-request-comment-trigger/pull/3 

If it looks okay, I'll merge and tag it as `v2` and `v2.1.0`

eliasmelgaco commented:
@shpingalet007  @shanegenschaw  for me your PRs are perfect. I would love to use both changes in the Khan project (main project)

shpingalet007 commented:
@eliasmelgaco okay, my PR was merged on [shanegenschaw/pull-request-comment-trigger](https://github.com/shanegenschaw/pull-request-comment-trigger). Feel free to leave your feedback here, 🙂 I will use the opportunity to fix any found issues on this PR and push updates to the fork 👍

shanegenschaw commented:
The work from @shpingalet007 has been merged and [published](https://github.com/marketplace/actions/pull-request-comment-trigger-fork) as `v2` and `v2.1.0`

shpingalet007 commented:
@shanegenschaw I removed `prefixOnly` flag. It is very unstable and difficult to make it so. Updated the PR, you can update your fork too if you want to.

shpingalet007 committed:
Remove prefixOnly

Trigger in middle of comment works very unstable. Removing prefixOnly and related functionality

shanegenschaw commented:
> @shanegenschaw I removed prefixOnly flag. It is very unstable and difficult to make it so. Updated the PR, you can update your fork too if you want to.

@shpingalet007 I've merged your commit into a new branch/PR, if you think it looks good: https://github.com/shanegenschaw/pull-request-comment-trigger/pull/6
